csharp/src/Ice/OutputStream-Class.cs

```diff
@@ -216,13 +236,22 @@ public void WriteTaggedClass(int tag, AnyClass? v)
             if (v != null)
             {
                 WriteTaggedParamHeader(tag, EncodingDefinitions.TagFormat.Class);
-                WriteClass(v);
+
+                // We cannot use formal type optimization for a tagged class parameter, because the recipient may not
+                // know the tag at all.
+                // Formal type optimization is never used for data members in sliced format.
```

Comment:
It's not totally clear to me what this comment refers to in the context of this method.

Reply:
Updated comment.